<a href="https://colab.research.google.com/github/ricomwas/PandaswithPython/blob/master/pandas_project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import requests

download_url = "https://raw.githubusercontent.com/fivethirtyeight/data/master/nba-elo/nbaallelo.csv"
target_csv_path = "nba_all_elo.csv"

response = requests.get(download_url)
response.raise_for_status()
with open(target_csv_path, "wb") as f:
  f.write(response.content)
print("Download ready")

Download ready


In [3]:
import pandas as pd

nba = pd.read_csv('nba_all_elo.csv')
type(nba)

pandas.core.frame.DataFrame

This is to determine the no of rows

In [4]:
len(nba)

126314

Checking your data dimensionality

In [5]:
nba.shape

(126314, 23)

Displaying the 1st 5 rows

In [6]:
nba.head()

gameorder       game_id lg_id  ...  game_result  forecast notes
0          1  194611010TRH   NBA  ...            L  0.640065   NaN
1          1  194611010TRH   NBA  ...            W  0.359935   NaN
2          2  194611020CHS   NBA  ...            W  0.631101   NaN
3          2  194611020CHS   NBA  ...            L  0.368899   NaN
4          3  194611020DTF   NBA  ...            L  0.640065   NaN

[5 rows x 23 columns]

Displaying the last 3 rows

In [7]:
nba.tail(3)

gameorder       game_id lg_id  ...  game_result  forecast notes
126311      63156  201506140GSW   NBA  ...            L  0.234435   NaN
126312      63157  201506170CLE   NBA  ...            L  0.481450   NaN
126313      63157  201506170CLE   NBA  ...            W  0.518550   NaN

[3 rows x 23 columns]

Displaying different datatypes

In [8]:
nba.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 126314 entries, 0 to 126313
Data columns (total 23 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   gameorder      126314 non-null  int64  
 1   game_id        126314 non-null  object 
 2   lg_id          126314 non-null  object 
 3   _iscopy        126314 non-null  int64  
 4   year_id        126314 non-null  int64  
 5   date_game      126314 non-null  object 
 6   seasongame     126314 non-null  int64  
 7   is_playoffs    126314 non-null  int64  
 8   team_id        126314 non-null  object 
 9   fran_id        126314 non-null  object 
 10  pts            126314 non-null  int64  
 11  elo_i          126314 non-null  float64
 12  elo_n          126314 non-null  float64
 13  win_equiv      126314 non-null  float64
 14  opp_id         126314 non-null  object 
 15  opp_fran       126314 non-null  object 
 16  opp_pts        126314 non-null  int64  
 17  opp_elo_i      126314 non-nul

Descriptive statics for all numeric columns

In [9]:
nba.describe()

gameorder        _iscopy  ...      opp_elo_n       forecast
count  126314.000000  126314.000000  ...  126314.000000  126314.000000
mean    31579.000000       0.500000  ...    1495.236055       0.500000
std     18231.927643       0.500002  ...     112.461687       0.215252
min         1.000000       0.000000  ...    1085.774400       0.020447
25%     15790.000000       0.000000  ...    1416.994900       0.327989
50%     31579.000000       0.500000  ...    1500.954400       0.500000
75%     47368.000000       1.000000  ...    1576.291625       0.672011
max     63157.000000       1.000000  ...    1853.104500       0.979553

[8 rows x 13 columns]

Displaying of other data types i.e. columns with the datatype of ojects

In [10]:
import numpy as np
nba.describe(include = np.object)


game_id   lg_id  ... game_result           notes
count         126314  126314  ...      126314            5424
unique         63157       2  ...           2             231
top     196112120NYK     NBA  ...           L  at New York NY
freq               2  118016  ...       63157             440

[4 rows x 10 columns]

Examining how often specific values occur in a column: using column:"team_id"

In [12]:
nba["team_id"].value_counts()

BOS    5997
NYK    5769
LAL    5078
DET    4985
PHI    4533
       ... 
INJ      60
DTF      60
TRH      60
PIT      60
SDS      11
Name: team_id, Length: 104, dtype: int64

Using column: "franc_id"

In [14]:
nba["fran_id"].value_counts()

Lakers          6024
Celtics         5997
Knicks          5769
Warriors        5657
Pistons         5650
Sixers          5644
Hawks           5572
Kings           5475
Wizards         4582
Spurs           4309
Bulls           4307
Pacers          4227
Thunder         4178
Rockets         4154
Nuggets         4120
Nets            4106
Suns            4080
Bucks           4034
Trailblazers    3870
Cavaliers       3810
Clippers        3733
Jazz            3555
Mavericks       3013
Heat            2371
Pelicans        2254
Magic           2207
Timberwolves    2131
Grizzlies       1657
Raptors         1634
Hornets          894
Colonels         846
Squires          799
Spirits          777
Stars            756
Sounds           697
Baltimore        467
Floridians       440
Condors          430
Capitols         291
Olympians        282
Sails            274
Stags            260
Bombers          249
Steamrollers     168
Packers           72
Redskins          65
Rebels            63
Denver       

We have Los Angeles Lakers & Minneapolis Lakers as diplayed below

In [15]:
which_lakers = nba.loc[nba["fran_id"] == "Lakers", "team_id"].value_counts()
which_lakers

LAL    5078
MNL     946
Name: team_id, dtype: int64

Find out when Minneapolis lakers played 946 games as shown above

In [20]:
min_date = nba.loc[nba["team_id"] == "MNL", "date_game"].min()

min_date 

'1/1/1949'

In [27]:
max_date = nba.loc[nba["team_id"] == "MNL", "date_game"].max()
max_date

'4/9/1959'

In [24]:
nba.loc[nba["team_id"] == "MNL", "date_game"].agg(("min", "max"))

min    1/1/1949
max    4/9/1959
Name: date_game, dtype: object

Minneaopolis played game between the years 1949 and 1959

Why has Boston Celtics "BOS" played most games in the dataset?

Will analyze their history, and find out how many points the BOS have scored in all matches.

In [30]:
nba.loc[nba["team_id"] == "BOS"]

gameorder       game_id lg_id  ...  game_result  forecast notes
6               4  194611020PRO   NBA  ...            L  0.359935   NaN
14              8  194611050BOS   NBA  ...            L  0.620204   NaN
28             15  194611090DTF   NBA  ...            L  0.386818   NaN
40             21  194611130CHS   NBA  ...            L  0.291740   NaN
46             24  194611140STB   NBA  ...            L  0.313890   NaN
...           ...           ...   ...  ...          ...       ...   ...
126141      63071  201504150MIL   NBA  ...            W  0.448084   NaN
126162      63082  201504190CLE   NBA  ...            L  0.263276   NaN
126173      63087  201504210CLE   NBA  ...            L  0.250867   NaN
126185      63093  201504230BOS   NBA  ...            L  0.502943   NaN
126205      63103  201504260BOS   NBA  ...            L  0.476631   NaN

[5997 rows x 23 columns]

Points scored by BOS in all matches

In [31]:
nba.loc[nba["team_id"] == "BOS", "pts"].sum()

626484